### PR TITLE
Volume and protection policy bug fix

### PR DESCRIPTION
**Pull request by prajwalpatil25**
# Description

1. Added update support for app_type and app_type_other attributes of volume.
2. Added error handling for logical_unit_number, appliance_id, appliance_name and sector_size attributes for modification is not supported.
3. Modified protectionPolicy reference variable to variable as per latest changes on gopowerstore side.
4. Modified protection policy acceptance tests to make use of environment variables instead of literal values of snapshot rule and replication rule.
5. Modified volume data source acceptance tests to make use of environment variables instead of literal values of volume id and volume name.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Acceptance tests

```
PASS
coverage: 80.7% of statements
ok      terraform-provider-powerstore/powerstore        392.108s        coverage: 80.7% of statements

```

### CODE REVIEW

**Review by AnikaAgiwal2711 — DISMISSED** [review]

**Comment by AnikaAgiwal2711** [comment]
Sorry , By mistakenly clicked on approve without reviewing it completely

File: powerstore/resource_protection_policy_test.go [hunk 1]
```diff
@@ -25,8 +25,8 @@ func TestAccProtectionPolicy_Create(t *testing.T) {
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr("powerstore_protectionpolicy.test", "name", "protectionpolicy_acc_new"),
 					resource.TestCheckResourceAttr("powerstore_protectionpolicy.test", "description", "Test CreateProtectionPolicy"),
-					resource.TestCheckResourceAttr("powerstore_protectionpolicy.test", "replication_rule_ids.0", "5d45b173-9a85-473e-8ab8-e107f8b8085e"),
-					resource.TestCheckResourceAttr("powerstore_protectionpolicy.test", "snapshot_rule_ids.0", "4be81573-c0e6-4956-a32f-a0e396a9b86d"),
+					resource.TestCheckResourceAttr("powerstore_protectionpolicy.test", "replication_rule_ids.0", replicationRuleID),
+					resource.TestCheckResourceAttr("powerstore_protectionpolicy.test", "snapshot_rule_ids.0", snapshotRuleID),
 				),
 			},
```
Comment (AnikaAgiwal2711) [inline powerstore/resource_protection_policy_test.go:31]: Can you point me what exactly you've done to achieve this **_"Modified protectionPolicy reference variable to variable as per latest changes on gopowerstore side."_** (stated in the description)

File: powerstore/volume_helper.go [hunk 2]
```diff
@@ -287,9 +299,11 @@ func modifyVolume(planVol models.Volume, valInBytes int64, volID string, client
 	vgModify := &pstore.VolumeModify{
 		Name:                planVol.Name.ValueString(),
 		Size:                valInBytes,
-		ProtectionPolicyID:  &protectionPolicy,
+		ProtectionPolicyID:  protectionPolicy,
```
Comment (prajwalpatil25) [inline powerstore/volume_helper.go:305]: @AnikaAgiwal2711 this is the change where "&protectionPolicy" is changed to "protectionPolicy" as it updated from a pointer to normal variable on gopowerstore side through https://github.com/dell/gopowerstore/pull/99/commits/1c5d1d09f5b464b78d13ed0d6811a02f8c4e4504

Comment (AnikaAgiwal2711) [reply powerstore/volume_helper.go:305]: Thanks

**Comment by AnikaAgiwal2711** [comment]
Looks good to me . I've already approved

**Review by AnikaAgiwal2711 — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**